hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,26 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    String database = identifier.namespace().level(0);
+    try {
+      Table table = clients.run(client -> client.getTable(database, identifier.name()));
+      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, identifier));
+      return true;
+    } catch (NoSuchTableException | NoSuchObjectException e) {
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to check table existence of " + identifier, e);
+    } catch (InterruptedException e) {
```

Comment:
Nit: do we need to do '      Thread.currentThread().interrupt();' as in other parts of the code?

Reply:
thank you added like others